consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/LocalNodeIsProposerTest.java

```diff
@@ -0,0 +1,119 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.tests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.pantheon.consensus.ibft.support.MessageReceptionHelpers.assertPeersReceivedExactly;
+import static tech.pegasys.pantheon.consensus.ibft.support.MessageReceptionHelpers.assertPeersReceivedNoMessages;
+
+import tech.pegasys.pantheon.consensus.ibft.ConsensusRoundIdentifier;
+import tech.pegasys.pantheon.consensus.ibft.IbftBlockHashing;
+import tech.pegasys.pantheon.consensus.ibft.IbftExtraData;
+import tech.pegasys.pantheon.consensus.ibft.ibftevent.BlockTimerExpiry;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.CommitPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.MessageFactory;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.ProposalPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.SignedData;
+import tech.pegasys.pantheon.consensus.ibft.support.RoundSpecificNodeRoles;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContext;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContextFactory;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.crypto.SECP256K1.Signature;
+import tech.pegasys.pantheon.ethereum.core.Block;
+
+import java.time.Clock;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * These tests assume the basic function of the Ibft Round State Machine has been proven
+ * via the LocalNodeIsNotProposerTest.
+ */
+public class LocalNodeIsProposerTest {
+
+
+  private final long blockTimeStamp = 100;
+  private final Clock mockBlockCreationTime = mock(Clock.class);
+
+  // Local node will propose the first block
+  private final TestContext context =
+      TestContextFactory.createTestEnvironment(4, 1, mockBlockCreationTime);
+  private final ConsensusRoundIdentifier roundId = new ConsensusRoundIdentifier(1, 0);
+  private final RoundSpecificNodeRoles roles = context.getRoundSpecificRoles(roundId);
+
+  private final MessageFactory localNodeMessageFactory = context.getLocalNodeMessageFactory();
+
+  private Block expectedProposedBlock;
+  private SignedData<ProposalPayload> expectedTxProposal;
+  private SignedData<CommitPayload> expectedTxCommit;
+
+  @Before
+  public void setup() {
+    when(mockBlockCreationTime.millis()).thenReturn(blockTimeStamp * 1000);
+    expectedProposedBlock = context.createBlockForProposal(0, blockTimeStamp);
+    expectedTxProposal =
+        localNodeMessageFactory.createSignedProposalPayload(roundId, expectedProposedBlock);
+
+    final IbftExtraData extraData =
```

Comment:
perhaps this could be moved into a common test util somewhere we are probably going to do this in many of the nodes. or perhaps just inside the createSignedCommitPayload using the proposedBlock as an arg to it?